File: 3주차-투포인터/w3-Yoon/BOJ_3273.java
import java.util.*;

import java.io.*;

 

public class Main{

​    public static void main(String[] args) throws IOException {

​        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

​        int n = Integer.parseInt(br.readLine());

​        int[] input = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).sorted().toArray();

​        int target = Integer.parseInt(br.readLine()), cnt = 0;

​        int start = 0, end = n-1;

​        

​        while(start < end){

​            int sum = input[start] + input[end];

​            if(sum == target){

​                end--; start++;

​                cnt++;

​            }else if(sum>target)

​                end--;

​            else

​                start++;

​        }

​        System.out.println(cnt);

​    }

}

File: 3주차-투포인터/w3-Yoon/BOJ_3273.py
import sys

read = sys.stdin.readline()

n=int(read())

arr=list(map(int, read(),split()))

x=int(read())

arr.sort()

start=0

end=n-1

cnt=0



while start < end:

​    sum = arr[start]+arr[end]

​    if sum == x:

​        cnt+=1

​    if sum < x:

​        start+=1

​    else:

​        end-=1



print(cnt)